aerial_robot_control/src/control/pose_linear_controller.cpp

```diff
@@ -328,12 +329,16 @@ namespace aerial_robot_control
     pid_msg_.yaw.err_p = err_yaw;
     pid_msg_.yaw.target_d = target_omega_.z();
     pid_msg_.yaw.err_d = target_omega_.z() - omega_.z();
+
+    yaw_from_pc_msg_.pc_yaw = rpy_.z();
+    yaw_from_pc_msg_.target_yaw = target_rpy_.z();
   }
 
   void PoseLinearController::sendCmd()
   {
     /* ros publish */
     pid_pub_.publish(pid_msg_);
+    yaw_from_pc_pub_.publish(yaw_from_pc_msg_);
```

Comment:
I recommend to create a new control class for MRAC called `AdaptiveController`, which is inherited from `PoseLinearController`, and merge your trigger python script into this new controller class